src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.ValidateFormatString
+{
+    internal abstract class AbstractValidateFormatStringDiagnosticAnalyzer<TSyntaxKind> :
+        DiagnosticAnalyzer
```

Comment:
I'm ok with that.  Note: why doesn't this have an option?  I'm thinking about the user that wants us to disable all this stuff.  Is it because we think this warning is pretty much always appropriate to show the user?  If so, i'm ok with making it non-disableable.